### PR TITLE
Migrating to hardhat

### PR DESCRIPTION
**Pull request by aalavandhan**
Setup project with latest deps

### CODE REVIEW

File: contracts/IStaking.sol [hunk 1]
```diff
@@ -1,24 +1,15 @@
...
  * @title Staking interface, as defined by EIP-900.
  * @dev https://github.com/ethereum/EIPs/blob/master/EIPS/eip-900.md
  */
-contract IStaking {
-    event Staked(address indexed user, uint256 amount, uint256 total, bytes data);
-    event Unstaked(address indexed user, uint256 amount, uint256 total, bytes data);
-
+interface IStaking {
     function stake(uint256 amount, bytes calldata data) external;
-    function stakeFor(address user, uint256 amount, bytes calldata data) external;
```
Comment (brandoniles) [inline contracts/IStaking.sol:12]: Any reason you removed the parameter name?

File: contracts/_utils/SafeMathCompatibility.sol [hunk 2]
```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+// NOTE: Adding an intermediate library to support older version of safemath.
```
Comment (brandoniles) [inline contracts/_utils/SafeMathCompatibility.sol:4]: Thinking of this as an intermediary, then replacing with the more basic math ops later?

File: deployments/TokenGeyser_v0.9.0_flattened.sol [hunk 3]
```diff
@@ -1,4 +1,3 @@
-
```
Comment (brandoniles) [inline deployments/TokenGeyser_v0.9.0_flattened.sol:1]: I'd support getting rid of these deployments, if they're not needed for etherscan verification anymore.

File: contracts/_utils/SafeMathCompatibility.sol (hunk 2 above)
Comment (aalavandhan) [reply contracts/_utils/SafeMathCompatibility.sol:4]: yes

File: contracts/IStaking.sol (hunk 1 above)
Comment (aalavandhan) [reply contracts/IStaking.sol:12]: hmm didn't mean to remove here. Adding it back

**Review by brandoniles — APPROVED** [review]

**Pull request merged**